.github/workflows/sonarcloud.yml

```
@@ -32,7 +32,13 @@ jobs:
-      - name: Run Tests and Analyze
+      - name: Run Tests and Analyze (All Microservices)
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: ./gradlew test jacocoTestReport sonarqube
+        run: |
+          for service in backend/*; do
+            if [ -f "$service/build.gradle" ]; then
+              echo "Analyzing $service"
+              (cd "$service" && ../../gradlew test jacocoTestReport sonarqube)
+            fi
+          done
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)